plugins/outputs/loki/loki.go

```diff
@@ -0,0 +1,217 @@
+package loki
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+const (
+	defaultEndpoint      = "/loki/api/v1/push"
+	defaultClientTimeout = 5 * time.Second
+	defaultFieldLine     = "line"
+)
+
+var sampleConfig = `
+  ## Connection timeout, defaults to "5s" if not set.
+  timeout = "5s"
+
+  ## The URL of Loki
+  # url = "https://loki.domain.tld"
```

Comment:
Please uncomment the `url` as it is required to be set by the user!

Reply:
indeed